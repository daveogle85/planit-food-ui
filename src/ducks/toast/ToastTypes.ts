export enum FeedbackStatus {
  ERROR = 'Error',
  WARN = 'Warn',
  INFO = 'Info',
  DISABLED = 'Disabled',
  HIDDEN = 'Hidden',
}

export type ToastState = {
  status: FeedbackStatus;
  message?: string;
};
