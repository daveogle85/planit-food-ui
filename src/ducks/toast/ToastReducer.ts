import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { RootState } from '..';
import { ToastState, FeedbackStatus } from './ToastTypes';

const initialState: ToastState = {
  status: FeedbackStatus.HIDDEN,
};

const toastSlice = createSlice({
  name: 'toast',
  initialState,
  reducers: {
    setToastState(state, action: PayloadAction<ToastState>) {
      state.status = action.payload.status;
      state.message = action.payload.message;
    },
  },
});

export const { setToastState } = toastSlice.actions;
export const toastSelectors = {
  selectToastState: (state: RootState) => state.toast,
};

export default toastSlice.reducer;
