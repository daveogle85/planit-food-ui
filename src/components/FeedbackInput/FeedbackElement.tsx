import classNames from 'classnames';
import React from 'react';

/** @jsx jsx */
import { jsx } from '@emotion/core';

import { FeedbackStatus } from '../../ducks/toast/ToastTypes';
import { FeedbackElementProps } from './FeedbackElementTypes';
import { styledFeedbackInput } from './StyledFeedbackInput';

function FeedbackElement(props: React.PropsWithChildren<FeedbackElementProps>) {
  const { children, className, state } = props;
  const hidden = state.status === FeedbackStatus.HIDDEN;
  return (
    <div
      className={classNames(className, {
        'feedback-hidden': hidden,
      })}
      css={styledFeedbackInput(state.status, state.message)}
    >
      {children}
      {!hidden && <div className="feedback-icon">i</div>}
    </div>
  );
}

export default (FeedbackElement as unknown) as React.FC<FeedbackElementProps>;
