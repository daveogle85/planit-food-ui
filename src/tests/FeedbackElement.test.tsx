import serializer from 'jest-emotion';
import React from 'react';
import renderer from 'react-test-renderer';
import FeedbackElement from '../components/FeedbackInput/FeedbackElement';
import { FeedbackStatus } from '../ducks/toast/ToastTypes';

expect.addSnapshotSerializer(serializer);

describe('FeedbackElement', () => {
  it('hidden', () => {
    const view = renderer
      .create(
        <FeedbackElement
          state={{
            status: FeedbackStatus.HIDDEN,
          }}
        >
          <div>Hidden</div>
        </FeedbackElement>
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it('error', () => {
    const view = renderer
      .create(
        <FeedbackElement
          state={{
            status: FeedbackStatus.ERROR,
            message: 'An Error Message',
          }}
        >
          <div>Error</div>
        </FeedbackElement>
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it('warn', () => {
    const view = renderer
      .create(
        <FeedbackElement
          state={{
            status: FeedbackStatus.WARN,
            message: 'A Warning message',
          }}
        >
          <div>Warn</div>
        </FeedbackElement>
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it('info', () => {
    const view = renderer
      .create(
        <FeedbackElement
          state={{
            status: FeedbackStatus.INFO,
            message: 'An Information message',
          }}
        >
          <div>Info</div>
        </FeedbackElement>
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });

  it('disable', () => {
    const view = renderer
      .create(
        <FeedbackElement
          state={{
            status: FeedbackStatus.DISABLED,
            message: 'A disabled message',
          }}
        >
          <div>Disabled</div>
        </FeedbackElement>
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });
});
