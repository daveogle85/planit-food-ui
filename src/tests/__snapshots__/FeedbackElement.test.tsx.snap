// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackElement disable 1`] = `
.emotion-0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0:not(.feedback-hidden) > *:first-child {
  padding-right: 25px;
  border: 2px solid #9b9797;
  border-radius: 0.3em;
  height: calc(100% - 6px);
}

.emotion-0 .feedback-icon {
  width: 13px;
  height: 13px;
  position: absolute;
  right: 5px;
  font-size: 12px;
  font-weight: 700;
  color: #9b9797;
  border-radius: 2em;
  border: 2px solid #9b9797;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.emotion-0 .feedback-icon:after {
  content: 'A disabled message';
  display: none;
  position: absolute;
  white-space: nowrap;
  border: 1px solid #9b9797;
  border-radius: 0.3em;
  padding: 2px;
  bottom: 150%;
  background: white;
  opacity: 0.9;
  font-size: 8px;
}

.emotion-0 .feedback-icon:hover:after {
  display: block;
}

<div
  className=" emotion-0"
>
  <div>
    Disabled
  </div>
  <div
    className="feedback-icon"
  >
    i
  </div>
</div>
`;

exports[`FeedbackElement error 1`] = `
.emotion-0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0:not(.feedback-hidden) > *:first-child {
  padding-right: 25px;
  border: 2px solid #cc0000;
  border-radius: 0.3em;
  height: calc(100% - 6px);
}

.emotion-0 .feedback-icon {
  width: 13px;
  height: 13px;
  position: absolute;
  right: 5px;
  font-size: 12px;
  font-weight: 700;
  color: #cc0000;
  border-radius: 2em;
  border: 2px solid #cc0000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.emotion-0 .feedback-icon:after {
  content: 'An Error Message';
  display: none;
  position: absolute;
  white-space: nowrap;
  border: 1px solid #cc0000;
  border-radius: 0.3em;
  padding: 2px;
  bottom: 150%;
  background: white;
  opacity: 0.9;
  font-size: 8px;
}

.emotion-0 .feedback-icon:hover:after {
  display: block;
}

<div
  className=" emotion-0"
>
  <div>
    Error
  </div>
  <div
    className="feedback-icon"
  >
    i
  </div>
</div>
`;

exports[`FeedbackElement hidden 1`] = `
.emotion-0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0:not(.feedback-hidden) > *:first-child {
  padding-right: 25px;
  border: 2px solid #9b9797;
  border-radius: 0.3em;
  height: calc(100% - 6px);
}

.emotion-0 .feedback-icon {
  width: 13px;
  height: 13px;
  position: absolute;
  right: 5px;
  font-size: 12px;
  font-weight: 700;
  color: #9b9797;
  border-radius: 2em;
  border: 2px solid #9b9797;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.emotion-0 .feedback-icon:after {
  content: '';
  display: none;
  position: absolute;
  white-space: nowrap;
  border: 1px solid #9b9797;
  border-radius: 0.3em;
  padding: 2px;
  bottom: 150%;
  background: white;
  opacity: 0.9;
  font-size: 8px;
}

.emotion-0 .feedback-icon:hover:after {
  display: block;
}

<div
  className="feedback-hidden emotion-0"
>
  <div>
    Hidden
  </div>
</div>
`;

exports[`FeedbackElement info 1`] = `
.emotion-0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0:not(.feedback-hidden) > *:first-child {
  padding-right: 25px;
  border: 2px solid #00703c;
  border-radius: 0.3em;
  height: calc(100% - 6px);
}

.emotion-0 .feedback-icon {
  width: 13px;
  height: 13px;
  position: absolute;
  right: 5px;
  font-size: 12px;
  font-weight: 700;
  color: #00703c;
  border-radius: 2em;
  border: 2px solid #00703c;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.emotion-0 .feedback-icon:after {
  content: 'An Information message';
  display: none;
  position: absolute;
  white-space: nowrap;
  border: 1px solid #00703c;
  border-radius: 0.3em;
  padding: 2px;
  bottom: 150%;
  background: white;
  opacity: 0.9;
  font-size: 8px;
}

.emotion-0 .feedback-icon:hover:after {
  display: block;
}

<div
  className=" emotion-0"
>
  <div>
    Info
  </div>
  <div
    className="feedback-icon"
  >
    i
  </div>
</div>
`;

exports[`FeedbackElement warn 1`] = `
.emotion-0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0:not(.feedback-hidden) > *:first-child {
  padding-right: 25px;
  border: 2px solid #ffbf00;
  border-radius: 0.3em;
  height: calc(100% - 6px);
}

.emotion-0 .feedback-icon {
  width: 13px;
  height: 13px;
  position: absolute;
  right: 5px;
  font-size: 12px;
  font-weight: 700;
  color: #ffbf00;
  border-radius: 2em;
  border: 2px solid #ffbf00;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.emotion-0 .feedback-icon:after {
  content: 'A Warning message';
  display: none;
  position: absolute;
  white-space: nowrap;
  border: 1px solid #ffbf00;
  border-radius: 0.3em;
  padding: 2px;
  bottom: 150%;
  background: white;
  opacity: 0.9;
  font-size: 8px;
}

.emotion-0 .feedback-icon:hover:after {
  display: block;
}

<div
  className=" emotion-0"
>
  <div>
    Warn
  </div>
  <div
    className="feedback-icon"
  >
    i
  </div>
</div>
`;
